test/Concurrency/actor_call_implicitly_async.swift

```diff
@@ -0,0 +1,177 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-concurrency
+// REQUIRES: concurrency
+
+actor class BankAccount {
+
+  private var curBalance : Int
+
+  private var accountHolder : String = "unknown"
+
+  // expected-note@+1 2 {{mutable state is only available within the actor instance}}
+  var owner : String {
+    get { accountHolder }
+    set { accountHolder = newValue }
+  }
+
+  init(initialDeposit : Int) {
+    curBalance = initialDeposit
+  }
+
+  // NOTE: this func is accessed through both async and sync calls.
+  // expected-note@+1 {{only asynchronous methods can be used outside the actor instance; do you want to add 'async'?}}
+  func balance() -> Int { return curBalance }
+
+  // expected-note@+1 {{only asynchronous methods can be used outside the actor instance; do you want to add 'async'?}}
+  func deposit(_ amount : Int) -> Int {
+    guard amount >= 0 else { return 0 }
+
+    curBalance = curBalance + amount
+    return curBalance
+  }
+
+  func canWithdraw(_ amount : Int) -> Bool { 
+    // call 'balance' from sync through self
+    return self.balance() >= amount
+  }
+
+  func testSelfBalance() async {
+    _ = await balance() // expected-warning {{no calls to 'async' functions occur within 'await' expression}}
+  }
+
+  // returns the amount actually withdrawn
+  func withdraw(_ amount : Int) -> Int {
+    guard canWithdraw(amount) else { return 0 }
+
+    curBalance = curBalance - amount
+    return amount
+  }
+
+  // returns the balance of this account following the transfer
+  func transferAll(from : BankAccount) async -> Int {
+    // call sync methods on another actor
+    let amountTaken = await from.withdraw(from.balance())
+    return deposit(amountTaken)
+  }
+
+  func greaterThan(other : BankAccount) async -> Bool {
+    return await balance() > other.balance()
+  }
+
+  func testTransactions() {
+    _ = deposit(withdraw(deposit(withdraw(balance()))))
+  }
+
+} // end actor class
+
+func someAsyncFunc() async {
+  let deposit1 = 120, deposit2 = 45
+  let a = BankAccount(initialDeposit: 0)
+  let b = BankAccount(initialDeposit: deposit2)
+
+  let _ = await a.deposit(deposit1)
+  let afterXfer = await a.transferAll(from: b)
+  let reportedBal = await a.balance()
+  
+  // check on account A
+  guard afterXfer == (deposit1 + deposit2) && afterXfer == reportedBal else {
+    print("BUG 1!")
+    return
+  }
+
+  // check on account B
+  guard await b.balance() == 0 else {
+    print("BUG 2!")
+    return
+  }
+
+  _ = await a.deposit(b.withdraw(a.deposit(b.withdraw(b.balance()))))
+
+  print("ok!")
+}
+
+
+//////////////////
+// check for appropriate error messages
+//////////////////
+
+extension BankAccount {
+  func totalBalance(including other: BankAccount) async -> Int {
+    return balance() 
+          + other.balance()  // expected-error{{call is 'async' but is not marked with 'await'}}
```

Comment:
We'll likely want to customize the diagnostic here, too, to say that the call to a synchronous function is implicitly `async`

Reply:
I added the "implicitly async" tidbit into the note that is produced alongside this error. Let me know if you think I should still customize this diagnostic.

Reply:
The "implicitly async" bit on the note is good, thanks!